src/core/thread/network_diagnostic.cpp

```diff
@@ -471,7 +444,7 @@ void NetworkDiagnostic::HandleDiagnosticGetQuery(Coap::Header &aHeader, Message
     // DIAG_GET.qry may be sent as a confirmable message.
     if (aHeader.GetType() == kCoapTypeConfirmable)
     {
-        SuccessOrExit(error = SendEmptyAck(aHeader, aMessageInfo));
+        SuccessOrExit(mCoapServer.SendEmptyResponse(aHeader, aMessageInfo));
```

Comment:
I think we still want to process the Query message, even if sending the response fails.